### PR TITLE
Fix/cr should activate disabled lag free

### PR DESCRIPTION
**Pull request by andreas-unleash**
- Adds change request option to activate disabled strategies UI
- Fixes Disable strategy bug (onSuggestDisable)

## About the changes

Closes #

### CODE REVIEW

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/MenuStrategyRemove/DisableEnableStrategyDialog/DisableEnableStrategyDialog.tsx [hunk 1]
```diff
@@ -28,7 +30,7 @@ export const DisableEnableStrategyDialog = ({
             if (disabled) {
                 onSuggestEnable();
             } else {
-                onSuggestEnable();
+                onSuggestDisable();
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/MenuStrategyRemove/DisableEnableStrategyDialog/DisableEnableStrategyDialog.tsx:33]: I'd add a test for The DisableEnableStrategyDialog replicating a failing scenario. I see the DisableEnableStrategyDialog doesn't have any tests.

Comment (kwasniew) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/MenuStrategyRemove/DisableEnableStrategyDialog/DisableEnableStrategyDialog.tsx:33]: I was doing similar component recently and ended up writing this test so you may use it as source of inspiration: https://github.com/Unleash/unleash/pull/3803/files#diff-519bb30b2e3d5f3c128a52eed4952c3c64bb5d945ac785d112c819f52892e0e3

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelEnvironmentSwitches/FeatureOverviewSidePanelEnvironmentSwitch/FeatureOverviewSidePanelEnvironmentSwitch.tsx [hunk 2]
```diff
@@ -155,30 +53,21 @@ export const FeatureOverviewSidePanelEnvironmentSwitch = ({
...
-                    }
-                    permission={UPDATE_FEATURE_ENVIRONMENT}
+                <FeatureToggleSwitch
+                    featureId={feature.name}
                     projectId={projectId}
-                    checked={enabled}
-                    onChange={toggleEnvironment}
-                    environmentId={name}
+                    environmentName={environment.name}
+                    onToggle={() => handleToggle()}
```
Comment (kwasniew) [inline frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewSidePanel/FeatureOverviewSidePanelEnvironmentSwitches/FeatureOverviewSidePanelEnvironmentSwitch/FeatureOverviewSidePanelEnvironmentSwitch.tsx:68]: anonymous function not necessary. onToggle={handleToggle}

File: frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx [hunk 3]
```diff
@@ -1,61 +1,223 @@
...
 }));
 
 interface IFeatureToggleSwitchProps {
-    featureName: string;
+    featureId: string;
     environmentName: string;
     projectId: string;
     value: boolean;
-    onToggle: (
+    showInfoBox?: () => void;
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx:28]: It's a very surprising API to me. What is showInfoBox?

File: frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx [hunk 4]
```diff
@@ -1,61 +1,223 @@
...
+    const [showEnabledDialog, setShowEnabledDialog] = useState(false);
+    const { feature } = useFeature(projectId, featureId);
+
+    const disabledStrategiesCount =
+        feature?.environments
+            .find(env => env.name === environmentName)
+            ?.strategies.filter(strategy => strategy.disabled).length ?? 0;
+
+    const callback = () => {
+        onToggle &&
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx:67]: onToggle?.(projectId, feature.name, environmentName, !isChecked);

Comment (kwasniew) [reply frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx:67]: The callback looses the semantics. All the other fns and variables are named specifically and callback is too generic. Once we introduce the refactoring I suggested above we may even get rid of the extra function

**Review by kwasniew — CHANGES_REQUESTED** [review]
A few comments inside. There's a lot of changes that look identical in the deleted code and new code. Has there been some move refactoring applied? I'd like to better understand which parts of the code are new changes so that I can spend more time reviewing those. Maybe inline comments would help.

File: frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx (hunk 3 above)
Comment (andreas-unleash) [reply frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx:28]: I know.  This was only passed in one the uses of the environment switch (in the feature overview sidepanel).  It is a dialog asking you to add a strategy before you can enable the environment (EnvironmentStrategyDialog)

File: frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/MenuStrategyRemove/DisableEnableStrategyDialog/DisableEnableStrategyDialog.tsx (hunk 1 above)
Comment (andreas-unleash) [reply frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/MenuStrategyRemove/DisableEnableStrategyDialog/DisableEnableStrategyDialog.tsx:33]: Will do

File: frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx [hunk 5]
```diff
@@ -1,61 +1,223 @@
...
+                key={key} // Prevent animation when archiving rows
+                data-testid={`TOGGLE-${key}`}
+            >
+                <PermissionSwitch
+                    tooltip={
+                        value
+                            ? `Disable feature in ${environmentName}`
+                            : `Enable feature in ${environmentName}`
+                    }
+                    checked={isChecked}
```
Comment (andreas-unleash) [inline frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx:191]: This line fixes the lagginess

File: frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx (hunk 3 above)
Comment (kwasniew) [reply frontend/src/component/project/Project/ProjectFeatureToggles/FeatureToggleSwitch/FeatureToggleSwitch.tsx:28]: I see it's only called on error. Maybe from the FeatureToggleSwitch perspective it should be onError?: () => void. FeatureToggleSwitch shouldn't care what happens in other part of the app. It should let the user of its API know that an error happened. We decouple from info box and avoid it leaking out into the FeatureToggleSwitch.

**Review by kwasniew — APPROVED** [review]

**Pull request merged**